packages/playwright/src/program.ts

```diff
@@ -218,7 +219,16 @@ async function runTests(args: string[], opts: { [key: string]: any }) {
     return;
   }
 
-  const runner = new Runner(config);
+  const lastRun = new LastRunReporter(config);
+  const runner = new Runner(config, lastRun);
+
+  if (opts.showWarnings) {
```

Comment:
If we go with this approach:
- handle `--show-warnings` before UI mode;
- perhaps do not filter projects, files, etc;
- throw when `--show-warnings` is used together with `--list`, `--ui` or `PWTEST_WATCH`.